src/ringct/rctSigs.cpp

```diff
@@ -1403,7 +1522,14 @@ namespace rct {
             offset += rv.p.rangeSigs.size();
           }
         }
-        if (!proofs.empty() && !verBulletproof(proofs))
+        if (!bpp_proofs.empty() && !verBulletproofPlus(bpp_proofs))
+        {
+          if (!waiter.wait())
```

Comment:
Probably because I was printing stuff from the results when debugging. Removed.

Reply:
Actually that's to avoid going past the scope while results (and possibly others) are still used by the threads. The thread dtor will wait, but it's a failsafe, you're supposed to wait first. I'll redo and add it to the other error path.